test/gurka/test_matrix.cc

```diff
@@ -916,3 +861,75 @@ TEST(StandAlone, HGVNoAccessPenalty) {
     }
   }
 }
+
+/************************************************************************ */
```

Comment:
Here we get to the relevant stuff: I added a test case that mimics the situation documented excellently by @radekvermirovsky in #4998 and make sure it now passes, with and without reverse connection checks.